eta/core/datasets.py

```diff
@@ -1582,111 +1625,118 @@ def _to_helper_list_video_objects(self, records):
+                ||np.dot(A, x) - b||
+            subject to:
+                x[i] is an element of [0, 1]
+
+        and different algorithms may be substituted in.
+
         Args:
-            target_remove: dictionary returned from Balancer._get_target_remove
-            helper_list: return value of Balancer._to_helper_list
+            A: the occurrence matrix computed in Balancer._get_occurrence_matrix
+            counts: (vector) the original counts for each value
+            target_count: (int) the target value
 
         Returns:
             a list of integer indices to keep
         '''
-        keep_idxs = list(range(len(helper_list)))
-        remove_idxs = []
-        score = self._get_score(target_remove, helper_list, remove_idxs)
-
-        # # ALGO1 - random
-        # import random
-        # random.seed(1)
-        # for _ in range(10000):
-        #     i = random.randrange(len(keep_idxs))
-        #     idx = keep_idxs[i]
-        #     score2 = self._get_score(
-        #         target_remove,
-        #         helper_list,
-        #         remove_idxs + [idx]
-        #     )
-        #     if score2 < score:
-        #         score = score2
-        #         remove_idxs.append(keep_idxs.pop(i))
-
-        # ALGO2 - greedy
-        while len(keep_idxs) > 0:
-            best_i = 0
-            best_score = self._get_score(
-                target_remove,
-                helper_list,
-                remove_idxs + [keep_idxs[best_i]]
-            )
-            for i in range(1, len(keep_idxs)):
-                cur_score = self._get_score(
-                    target_remove,
-                    helper_list,
-                    remove_idxs + [keep_idxs[i]]
-                )
-                if cur_score < best_score:
-                    best_score = cur_score
-                    best_i = i
-
-            if best_score >= score:
-                break
-            score = best_score
-            remove_idxs.append(keep_idxs.pop(best_i))
+        b = counts - target_count
 
-        return keep_idxs
+        # x = self._random(A, b)
+        x = self._greedy(A, b)
 
-    def _get_score(self, target_remove, helper_list, remove_idxs):
-        '''Compute the performance 'score' for a given set of indices to remove.
+        # print('counts: {}'.format(counts))
+        # print('target_count: {}'.format(target_count))
+        # print('b: {}'.format(b))
+        # print('dot(A,x): {}'.format(np.dot(A, x)))
+        # print('b - dot(A,x): {}'.format(b - np.dot(A, x)))
+        # print('output counts: {}'.format(counts - np.dot(A, x)))
+        # print('~~~~~~~~~~~')
+        # print(self._solution_score(b - np.dot(A, x)))
+
+        return np.where(x==0)[0]
+
+    def _random(self, A, b):
+        '''A random search algorithm for finding the indices to omit.
+
+        Args:
+            A: the occurrence matrix
+            b: the target vector to match
+
+        Returns:
+            x: the solution vector
```

Comment:
I suggest writing here also that `x[j] = 1` means you _omit_ that index, just because it keeps confusing me :)